### PR TITLE
Bug 1109592

### PR DESCRIPTION
**Pull request by tdz**

### CODE REVIEW

File: src/nfcd.cpp [hunk 1]
```diff
@@ -23,7 +26,96 @@
...
+        res = parseOpt_a(c, optarg);
+        break;
+      case 'h':
+        res = parseOpt_h(c, optarg);
+        break;
+      case '?':
+        res = parseOpt_QuestionMark(c, optarg);
+        break;
+      default:
+        res = -1; /* unknown option */
```
Comment (DimiL) [inline src/nfcd.cpp]: Indentation. two spaces before |case|

File: src/NfcIpcSocket.cpp [hunk 2]
```diff
@@ -90,34 +90,88 @@ int NfcIpcSocket::getListenSocket() {
...
+
+  return nfcdRw;
+}
+
 void NfcIpcSocket::setSocketListener(IpcSocketListener* listener) {
   mListener = listener;
 }
 
-void NfcIpcSocket::loop()
+void NfcIpcSocket::loop(const char* socketName)
```
Comment (DimiL) [inline src/NfcIpcSocket.cpp]: Recently i update a patch that function name will begin with upper case and all arguments will start with 'a', ex aSocketName to follow Mozilla C++ coding style.
Could you also help update the naming style, thanks

File: src/NfcIpcSocket.cpp [hunk 3]
```diff
@@ -90,34 +90,88 @@ int NfcIpcSocket::getListenSocket() {
...
+        nfcdConn = getListenSocket();
+        if (nfcdConn < 0) {
+          nanosleep(&mSleep_spec, &mSleep_spec_rem);
+          continue;
+        }
       }
-    }
 
-    nfcdRw = accept(nfcdConn, (struct sockaddr*)&peeraddr, &socklen);
+      nfcdRw = accept(nfcdConn, (struct sockaddr*)&peeraddr, &socklen);
```
Comment (DimiL) [inline src/NfcIpcSocket.cpp]: Nits, Indentation

File: src/NfcIpcSocket.cpp [hunk 4]
```diff
@@ -90,34 +90,88 @@ int NfcIpcSocket::getListenSocket() {
...
+    return -1;
+  }
+
+  struct sockaddr_un addr;
+  addr.sun_family = AF_UNIX;
+  addr.sun_path[0] = '\0'; /* abstract socket namespace */
+  memcpy(addr.sun_path + 1, socketName, len + 1);
+  socklen_t addrLen = offsetof(struct sockaddr_un, sun_path) + siz;
+
+  int nfcdRw = socket(AF_UNIX, SOCK_STREAM, 0);
```
Comment (DimiL) [inline src/NfcIpcSocket.cpp]: There is also a global static int nfcdRw variable in NfcIpcSocket.cpp, could you help either not declare nfcdRw here or remove the global static variable.

File: src/nfcd.cpp [hunk 5]
```diff
@@ -23,7 +26,96 @@
...
+
+int main(int argc, char *argv[]) {
+
+  struct Options options;
+  int res = options.parse(argc, argv);
+
+  if (res > 0)
+    return EXIT_SUCCESS;
+  else if (res < 0)
+    return EXIT_FAILURE;
```
Comment (DimiL) [inline src/nfcd.cpp]: add {}

File: src/nfcd.cpp [hunk 6]
```diff
@@ -23,7 +26,97 @@
...
+           "  -a    the network address\n"
+           "\n"
+           "The only supported address family is AF_UNIX with abstract\n"
+           "names. Not setting a network address will create a listen\n"
+           "socket.\n");
+
+    return 1;
+  }
+
+  int ParseOpt_QuestionMark(int aC, char* aarg)
```
Comment (DimiL) [inline src/nfcd.cpp]: aArg

File: src/NfcIpcSocket.cpp [hunk 7]
```diff
@@ -91,37 +90,91 @@ int NfcIpcSocket::GetListenSocket() {
   return nfcdConn;
 }
 
+int NfcIpcSocket::GetConnectedSocket(const char* aSocketName)
+{
+  size_t len = strlen(aSocketName);
+  if (len > (SIZE_MAX - 2)) {
```
Comment (allstarschh) [inline src/NfcIpcSocket.cpp]: what is 2 for ?

Comment (tdz) [inline src/NfcIpcSocket.cpp]: It represents the leading and trailing '\0' of the socket name. I'll add a constant.

**Pull request merged**